### PR TITLE
[GEOT-7267] force correct srs style property

### PR DESCRIPTION
**Pull request by NielsCharlier**
[GEOT-7267](https://osgeo-org.atlassian.net/browse/GEOT-7267) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

Geotools force OGC_HTTP_URL syntax on CRS that are east/north. This causes problems for INSPIRE and this cannot be resolved with configuration in geoserver, because it is hard coded and ignores everything else. I don't think this should happen in theory, but had to make this into a configurable property for backwards compatibility.
  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [x] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [x] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [x] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

File: modules/extension/xsd/xsd-gml2/src/test/java/org/geotools/gml2/GML3EncodingUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package org.geotools.gml2;
```
[the remark this answers is not in the document]
Comment (aaime) [reply modules/extension/xsd/xsd-gml2/src/test/java/org/geotools/gml2/GML3EncodingUtilsTest.java:18]: If I recall correctly, style checks do not apply to test packages, there were too many files to correct. Volunteers welcomed.

[the remark this answers is not in the document]
Comment (mprins) [reply modules/extension/xsd/xsd-gml2/src/test/java/org/geotools/gml2/GML3EncodingUtilsTest.java:18]: hmm spotless did check a test class:
```
Error:  Failed to execute goal com.diffplug.spotless:spotless-maven-plugin:2.20.0:check (default) on project gt-xsd-gml2: The following files had format violations:
Error:      src/test/java/org/geotools/gml2/GML3EncodingUtilsTest.java
Error:          @@ -14,7 +14,7 @@
Error:           ·*····MERCHANTABILITY·or·FITNESS·FOR·A·PARTICULAR·PURPOSE.··See·the·GNU
Error:           ·*····Lesser·General·Public·License·for·more·details.
Error:           ·*/
Error:          -·
Error:          +
Error:           package·org.geotools.gml2;
Error:           
Error:           import·static·org.junit.Assert.assertEquals;
Error:  Run 'mvn spotless:apply' to fix these violations.
Error:  -> [Help 1]
```

[the remark this answers is not in the document]
Comment (aaime) [reply modules/extension/xsd/xsd-gml2/src/test/java/org/geotools/gml2/GML3EncodingUtilsTest.java:18]: Yes, those are enabled... headers are checked by "checkstyle" not by spotless. That's what Jody meant by "style checks" I believe.

File: modules/library/metadata/src/main/java/org/geotools/util/factory/GeoTools.java [hunk 2]
```diff
@@ -186,6 +186,26 @@ private static Properties loadProperites(String resource) {
     public static final String FORCE_LONGITUDE_FIRST_AXIS_ORDER =
             "org.geotools.referencing.forceXY";
 
+    /**
+     * The {@linkplain System#getProperty(String) system property} key for the default value to be
+     * assigned to the {@link Hints# ENTITY_RESOLVER} hint.
+     *
+     * <p>GML Encoding in geotools forces http://www.opengis.net/gml/srs/epsg.xml# syntax on CRS
+     * that are East\North oriented or have no orientation. This property prevents this behaviour.
```
Comment (aaime) [inline modules/library/metadata/src/main/java/org/geotools/util/factory/GeoTools.java:194]: The ticket talks about one-dimensional CRS but the implementation seems to cover 2d CRSs as well.
Is the flag compatible with what declared at https://docs.geoserver.org/2.22.x/en/user/services/wfs/axis_order.html ?

File: modules/library/metadata/src/main/java/org/geotools/util/factory/GeoTools.java [hunk 3]
```diff
@@ -186,6 +186,26 @@ private static Properties loadProperites(String resource) {
     public static final String FORCE_LONGITUDE_FIRST_AXIS_ORDER =
             "org.geotools.referencing.forceXY";
 
+    /**
+     * The {@linkplain System#getProperty(String) system property} key for the default value to be
+     * assigned to the {@link Hints# ENTITY_RESOLVER} hint.
```
[the remark this answers is not in the document]
Comment (NielsCharlier) [reply modules/library/metadata/src/main/java/org/geotools/util/factory/GeoTools.java]: Fixed

**Comment by jodygarnett** [comment]
@NielsCharlier if you generate the expected URL ... that is good for INSPIRE but may still be confusing...

Q: But if GeoTools  *means* the forcedXY axis order used internally ... then the generated URL is misleading is not?

**Comment by NielsCharlier** [comment]
> @NielsCharlier if you generate the expected URL ... that is good for INSPIRE but may still be confusing...
> 
> Q: But if GeoTools _means_ the forcedXY axis order used internally ... then the generated URL is misleading is not?

@jodygarnett. 

1. As far as I can see, the default behaviour as described by the docs remains the same. Each of the GMLConfiguration have their own default syntax initialised like this:

    /** Srs name style to encode srsName URI's with */
    protected SrsSyntax srsSyntax = SrsSyntax.OGC_HTTP_URL;
    
 The problem is only is when the user explicitly configures the CRS syntax to something else on the WFS configuration page. I don't see what would be more confusing then geoserver ignoring what the user is trying to configure.

2. And even then, one has to explicitly turn the flag on for any behaviour to change, that is done as an additional backwards compatibility safety measure. I have to admit, I have given up hope to ever fully comprehend geotools/geoserver behaviour with respect to axis order. For that reason I would not dare to take the risk to make any changes without a flag or configuration explicitely needing to be turned on first.

**Comment by NielsCharlier** [comment]
@aaime thanks for the merge. The default value of the flag is false so without setting the flag the behaviour will stay the same.

**Pull request merged**